server/neptune/workflows/internal/deploy/revision/queue/deployer.go

```diff
@@ -86,12 +91,48 @@ func (p *Deployer) Deploy(ctx workflow.Context, requestedDeployment terraformWor
 		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", key.ErrKey, persistErr)
 	}
 
+	// rebase open PRs
+	// worker uses the returned error types to perform some follow up tasks, so instead of propagating the rebase error back, we configure
+	// infinite retries and log the error to ensure we rebase open PRs before deploying another change in this root
+	// ScheduleToCloseTimeout is configured to 8 hours to ensure we're not blocking deploys for this root if the issue persists
+	if rebaseErr := p.rebaseOpenPRsForRoot(ctx, requestedDeployment.Repo); rebaseErr != nil {
+		scope.Counter("rebase_open_prs_err").Inc(1)
+		logger.Error(ctx, "unable to rebase open PRs", key.ErrKey, rebaseErr)
+	}
+
 	// Count this as deployment as latest if it's not a PlanRejectionError which means it is a TerraformClientError
 	// We do this as a safety measure to avoid deploying out of order revision after a failed deploy since it could still
 	// mutate the state file
 	return latestDeployment, err
 }
 
+func (p *Deployer) rebaseOpenPRsForRoot(ctx workflow.Context, repo github.Repo) error {
+	// skip rebasing open PRs if the execution is still on the old workflow version
+	version := workflow.GetVersion(ctx, version.RebaseOpenPRs, workflow.DefaultVersion, 1)
+	if version == workflow.DefaultVersion {
+		return nil
+	}
+
+	// configure infinite retries and ScheduleToCloseTimeout to 8 hours to allow for the GH API Ratelimit to revive if we hit the ratelimit since it resets every hour
+	opts := workflow.GetActivityOptions(ctx)
+	opts.ScheduleToCloseTimeout = RebaseScheduleToCloseTimeout
+	opts.RetryPolicy = &temporal.RetryPolicy{
+		MaximumAttempts: 0,
+	}
```

Comment:
You don't need to specify this, this is the default value.  I also don't think this should be infinite since again it's not critical for the function of this workflow.  This can cause a thundering herd since we'll be doing a lot of gh api calls here and if gh is having an outage and we're hammering them across a ton of workflows, we'll probably have a rough time with our rate limits, let's just give it a best effort of 3 tries and bail out of it.

Reply:
If we configure max retries to 3 and all 3 retries happen within the span of an hour when we are out of API budget, we'd fail this activity without letting it auto resolve.  

I think if we configure the `ScheduleToCloseTimeout` to > 1hr for it to auto resolve with a higher backoff coefficient so that it retries not more than 3 times in an hour, we would get the best of both worlds. What do you think?